tree/dataframe/inc/ROOT/RDF/RVariation.hxx

```diff
@@ -61,91 +54,103 @@ bool ResultsSizeEq(const RVec<T> &results, std::size_t expected, std::size_t nCo
 }
 
 template <typename T>
-std::size_t GetNVariations(const RVec<RVec<T>> &results)
+std::size_t GetNVariations(const RVec<T> &results)
 {
-   assert(!results.empty());
-   return results[0].size();
+   return results.size();
 }
 
 template <typename T>
-std::size_t GetNVariations(const RVec<T> &results)
+void ResizeResults(ROOT::RVec<T> &results, std::size_t /*nCols*/, std::size_t nVariations)
 {
-   return results.size();
+   results.resize(nVariations);
 }
 
-template <typename RVec_t, typename Value_t = typename RVec_t::value_type>
-std::enable_if_t<!IsRVec<Value_t>::value, const std::type_info &> GetInnerValueType(std::size_t)
+// Assign into fLastResults[slot] without changing the addresses of its elements (we gave those addresses away in
+// GetValuePtr)
+template <typename T>
+void AssignResults(ROOT::RVec<T> &resStorage, ROOT::RVec<T> &&tmpResults)
 {
-   return typeid(Value_t);
+   const auto nVariations = resStorage.size(); // we have already checked that tmpResults has the same size
+
+   for (auto i = 0u; i < nVariations; ++i)
+      resStorage[i] = std::move(tmpResults[i]);
 }
 
-template <typename RVec_t, typename Value_t = typename RVec_t::value_type>
-std::enable_if_t<IsRVec<Value_t>::value, const std::type_info &> GetInnerValueType(std::size_t nCols)
+template <typename T>
+void *GetValuePtrHelper(ROOT::RVec<T> &v, std::size_t /*colIdx*/, std::size_t varIdx)
```

Comment:
The address returned here ends up being cast back to a `T*` and then passed to user callables as a (non-const) `T&`.
So in order to return a `const void *` here we would have to enforce that all user callables always take values by `const T&`. That's a backward-incompatible change and in some corner cases it might not be desirable, e.g. if the type of the column is `SomeComplexObject&` on which the user has to call a non-const method (maybe because the object comes from some legacy code and someone somewhen forgot to mark it const. maybe that method is `TH1::FindBin`).